platform/import-export/src/actions/ExportAsJSONAction.js

```diff
@@ -68,35 +76,38 @@ define([], function () {
      * @param {Object} parent
      */
     ExportAsJSONAction.prototype.write = function (parent) {
-
         this.calls++;
-        if (parent.hasCapability('composition')) {
-            parent.useCapability('composition')
+        var composition = this.openmct.composition.get(parent);
+
+        if (composition !== undefined) {
+            composition.load()
                 .then(function (children) {
                     children.forEach(function (child, index) {
                         // Only export if object is creatable
                         if (this.isCreatable(child)) {
                             // Prevents infinite export of self-contained objs
-                            if (!this.tree.hasOwnProperty(child.getId())) {
+                            if (!this.tree.hasOwnProperty(this.getId(child))) {
```

Comment:
Hi @akhenry 
I will suggest caching this.getId(child) on the top once to save the query for the next time :)
```
let childId = this.getId(child)

```
and then:
```
if (!this.tree.hasOwnProperty(childId) {
```